DetectorDescription/OfflineDBLoader/src/DDCoreToDDXMLOutput.cc

```diff
@@ -271,9 +271,9 @@ void DDCoreToDDXMLOutput::solid(const DDSolid& solid, std::ostream& xos) {
       std::vector<double> zs = rs.zscaleVec();
 
       xos << "<ExtrudedPolygon name=\"" << rs.toString() << "\"";
-      for (unsigned int i : x)
+      for (unsigned int i = 0; i < x.size(); ++i)
```

Comment:
@bsunanda sorry, why does HGCal geometry imply a difference between these two syntaxes?

Reply:
HGCal geometry uses extruded polygon and with that in the scenario, the code used to crash in the first syntax and produces the correct xml file in the second.

Reply:
The old code was assigning `i = x[0]`, then `i = x[1]`, etc. It would use `i` as the index in `x[i]` in line 275. That reference would probably be way out of bounds and could have caused the crash.

Reply:
ok, sorry, a trivial confusion between array index and element. Printing i instead of x[i] the range loop would be ok, but you need the index also for the other variable